### ISSUE TITLE
Introduce getOrComputeIfAbsent() variant for default constructor in ExtensionContext.Store

### ISSUE THREAD

**Issue by sbrannen** [junit-team/junit5#1110]
## Overview

After gaining experience with implementing extensions that wish to store a single object a given type using the type's default constructor, I have discovered that I always copy-n-paste the following code snippet to each extension implementation (where `X` is the type of object supported by the extension).

```java
X x = store.getOrComputeIfAbsent(X.class, k -> new X(), X.class);
```

This pattern can be seen in `@TrackLogRecords.Extension.getListener(ExtensionContext)` within our own test suite as well as in the [`@CaptureSystemOutput`](https://github.com/sbrannen/junit5-demo/blob/master/src/test/java/extensions/CaptureSystemOutput.java) extension that I created yesterday.

Since the `getOrComputeIfAbsent(X.class, k -> new X(), X.class)` pattern is so common, it would be nice to introduce a convenience method in `ExtensionContext.Store` with the following signature that does the same thing behind the scenes: `public <V> V getOrComputeIfAbsent(Class<V> type)`.

## Deliverables

- [x] Introduce `public <V> V getOrComputeIfAbsent(Class<V> type)` in `ExtensionContext.Store` that gets an object of type `type` (keyed by `type`) if present in the store and otherwise invokes the default constructor for `type` to generate the object.

**Comment by sbrannen** [comment]
_in progress_

**Comment by sbrannen** [comment]
Resolved in `master` in commit f052e3f09a1032c264e51bb7e6b99771a28a8c61.
